lib/rules/no-this-in-template-only-components.js

```diff
@@ -0,0 +1,80 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+
+const Rule = require('./base');
+
+function message(original, fixed) {
+  return (
+    `Usage of 'this' in path '${original}' is not allowed in a template-only component. ` +
+    `Use '${fixed}' if it is a named argument or create a component.js for this component.`
+  );
+}
+
+function hasContext(hbsFilePath) {
```

Comment:
Mind renaming this to `isTemplateOnlyComponent`? I could totally imagine extracting this into a shared utility in the future...